WatchDog/watchdog.c

```diff
@@ -0,0 +1,62 @@
+nt inPin = 7; //whatever the input pin is
+int outPin = 13; //output to tell other hardware if software failed
+int state = 0;   //output value, start LOW
+int lastState = 0; //to check for changing vals
+int changeCounter = 0; //just looking for changes
+int totalCounter = 0; //to check if we are changing fast enough
+int mainLoop = 0; //turn on once we know program is running
+
+//wait for X amount of signal changes (heart beats) within a short 
+//time period (to make sure power cycles or other transients doesn't 
+//accidentally turn it on)
+//no idea what these should be
+int waitPeriod = 15;
+int goal = 10; 
+
+
+void setup(){
+pinMode(outPin, OUTPUT);
+pinMode(inPin, INPUT);
+}
+
+void loop(){
+
```

Comment:
I suggest structuring all of the states in a single `if` or `case switch`. Store the state in an `int` from 0 to 3. It is easier to read which state you are in that way

Reply:
And structure it sequentially as well!